packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,274 @@
+  align-items: center;
+  cursor: pointer;
+  margin: var(--onyx-spacing-2xs);
+  height: 1.5rem;
+
+  &__input {
+    position: absolute;
+    opacity: 0;
+
+    &:checked + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-primary-500);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-themed-neutral-100);
+        transform: translateX(14px);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-primary-intense);
+        }
+      }
+    }
+
+    &:checked:disabled + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-primary-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-primary-soft);
+        }
+      }
+    }
+
+    &:disabled + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-neutral-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-neutral-300);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--invalid + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-danger-200);
+        border: var(--onyx-1px-in-rem) solid var(--onyx-color-base-danger-500);
+        height: 1.375rem;
+        width: 2.25rem;
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-danger-500);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--invalid:checked + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-danger-500);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-danger-intense);
+        }
+      }
+    }
+
+    &--readonly + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-neutral-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-neutral-300);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-neutral-inverted);
+        }
+      }
+    }
+
+    &--readonly:checked + .onyx-switch__container {
+      .onyx-switch__track {
+        background-color: var(--onyx-color-base-primary-200);
+      }
+
+      .onyx-switch__position {
+        background-color: var(--onyx-color-base-background-blank);
+
+        .onyx-switch__icon {
+          color: var(--onyx-color-text-icons-primary-soft);
+        }
+      }
+    }
+  }
+
+  &__container {
+    position: relative;
```

Comment:
Please try to avoid `position: relative` and `position: absolute` if they are not absolutely necessary